From 3b307fe0f0c8ea10c66f43dc5133d22f5e86c315 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Mon, 12 Apr 2021 10:36:44 +0530
Subject: [PATCH] Revert "FROMGIT: drm/i915/display: Defer initial modeset
 until after GGTT is initialised"

This reverts commit 2221eaf1bb8c7724c5dca7091d76e10ebbd3e0d0.

Commit reverted as it causes Guest boot issue in VM setup.

Tracked-On: OAM-96764
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 drivers/gpu/drm/i915/display/intel_display.c | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/drivers/gpu/drm/i915/display/intel_display.c b/drivers/gpu/drm/i915/display/intel_display.c
index 9f4c1249eca7..91b6f73c8fe5 100644
--- a/drivers/gpu/drm/i915/display/intel_display.c
+++ b/drivers/gpu/drm/i915/display/intel_display.c
@@ -18045,6 +18045,16 @@ int intel_modeset_init_nogem(struct drm_i915_private *i915)
 	if (!HAS_GMCH(i915))
 		sanitize_watermarks(i915);
 
+	/*
+	 * Force all active planes to recompute their states. So that on
+	 * mode_setcrtc after probe, all the intel_plane_state variables
+	 * are already calculated and there is no assert_plane warnings
+	 * during bootup.
+	 */
+	ret = intel_initial_commit(dev);
+	if (ret)
+		drm_dbg_kms(&i915->drm, "Initial commit in probe failed.\n");
+
 	return 0;
 }
 
@@ -18056,16 +18066,6 @@ int intel_modeset_init(struct drm_i915_private *i915)
 	if (!HAS_DISPLAY(i915))
 		return 0;
 
-	/*
-	 * Force all active planes to recompute their states. So that on
-	 * mode_setcrtc after probe, all the intel_plane_state variables
-	 * are already calculated and there is no assert_plane warnings
-	 * during bootup.
-	 */
-	ret = intel_initial_commit(&i915->drm);
-	if (ret)
-		return ret;
-
 	intel_overlay_setup(i915);
 
 	ret = intel_fbdev_init(&i915->drm);
-- 
2.17.1

